### PR TITLE
Dualstack support with AWS CCM

### PR DESCRIPTION
**Pull request by sachintiptur**
Signed-off-by: Sachin Tiptur <sachin@kubermatic.com>

**What this PR does / why we need it**:
Support dual stack IPs with AWS CCM.
**Which issue(s) this PR fixes**:

Fixes # [11171](https://github.com/kubermatic/kubermatic/issues/11171)

**What type of PR is this?**

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Add support for dual-stack node IPs with AWS CCM
```

**Documentation**:

```documentation
Doc changes are part of https://github.com/kubermatic/docs/pull/1254
```

### CODE REVIEW

File: pkg/resources/cloudconfig/cloudconfig.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ func CloudConfig(
 	cloud := cluster.Spec.Cloud
 	switch {
 	case cloud.AWS != nil:
+		DualstackIPFamilies := []string{"ipv6", "ipv4"}
```
Comment (rastislavs) [inline pkg/resources/cloudconfig/cloudconfig.go]: should be in reversed order: `"ipv4", "ipv6"` - we do not support IPv6 as the primary IP family in KKP yet

**Comment by ahmedwaleedmalik** [comment]
@xrstf `pre-kubermatic-test-integration` needs an override; failing because of vSphere.

**Comment by xrstf** [comment]
/hold

AWS is full.

**Pull request merged**